### ISSUE TITLE
[iOS] setStyleJSON

### ISSUE THREAD

**Issue by robipresotto** [mapbox/mapbox-gl-native#954]
I can not parse the JSON Style by setStyleJSON... (the default style - basic-v7.json)
I'm trying something like that,

- (NSString *)jsonStyleFromFilePath:(NSString *)filePath {

    NSError *deserializingError;
    NSData *content = [[NSData alloc] initWithContentsOfFile:filePath];
    NSString *object = [NSJSONSerialization JSONObjectWithData:content options:NSJSONReadingMutableLeaves error:&deserializingError];
    return object;
}

NSString *filePath = [[NSBundle mainBundle] pathForResource:@"default" ofType:@"json"];
[self.mapView setStyleJSON:jsonStyleFromFilePath:filePath];

On Debug the error msg
2015-03-06 14:04:50.868 Embark[11143:1687668] -[__NSCFDictionary cStringUsingEncoding:]: unrecognized selector sent to instance 0x7fbad30e3100

**Comment by friedbunny** [comment]
One issue here is that `setStyleJSON` takes the JSON as `NSString`, and you can't cast `NSJSONSerialization` to a string (as it's [meant to serialize JSON into an array or dictionary](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSJSONSerialization_Class/index.html)).

The style API is still being decided and implemented (#927 and so on), but here's a way that currently works to initialize a map with custom `style.json`:

```objective-c
- (void)viewDidLoad {
  // ... skipped all the other init like access token, etc

  NSString *styleJsonPath = [[NSBundle mainBundle] pathForResource:@"styles/basic-v7" ofType:@"json"];
  NSString *styleJson = [NSString stringWithContentsOfFile:styleJsonPath encoding:NSUTF8StringEncoding error:NULL];
    
  self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleJSON:styleJson accessToken:accessToken];

  // ... and the rest of your setup
}
```

**Comment by robipresotto** [comment]
Works like a charm! :+1: 
Thank you so much! :)))

Do you know the best way to set the custom sprites?
